Fix future unknown `Event` variant backwards compatibility

TheBlueMatt opened the pull request:
In 8ffc2d1742ff1171a87b0410b21cbbd557ff8247, in 0.0.100, we added
a backwards compatibility feature to the reading of `Event`s - if
the type was unknown and odd, we'd simply ignore the event and
treat it as no event. However, we failed to read the
length-prefixed TLV stream when doing so, resulting in us reading
some of the skipped-event data as the next event or other data in
the ChannelManager.

We fix this by reading the varint length prefix written, then
skipping that many bytes when we come across an unknown odd event
type.

We really need a good test framework for reading new or old data with old or new channelmanagers/etc, but for 0.0.101 it doesn't make sense to try to build one.

jkczyz commented on lightning/src/util/events.rs:
```diff
@@ -379,7 +384,19 @@ impl MaybeReadable for Event {
+				// If the event is of unknown type, assume it was written with `write_tlv_fields`,
+				// which prefixes the whole thing with a length BigSize. Because the event is
+				// odd-type unknown, we should treat it as `Ok(None)` even if it has some TLV
+				// fields which are event. Thus, we avoid using `read_tlv_fields` and simply read
```
s/event/even
s/which/that

TheBlueMatt commented:
Squashed and rebased with no other changes.

grammar fixup commit was:
```
94875fade6a1215ccda3bcc5633e30ab9405f1d2
--- a/lightning/src/util/events.rs
+++ b/lightning/src/util/events.rs
@@ -392,3 +392,3 @@ impl MaybeReadable for Event {
                                // odd-type unknown, we should treat it as `Ok(None)` even if it has some TLV
-                               // fields which are event. Thus, we avoid using `read_tlv_fields` and simply read
+                               // fields that are even. Thus, we avoid using `read_tlv_fields` and simply read
                                // exactly the number of bytes specified, ignoring them entirely.
```

Range-diff is `git range-diff 801d6e5256d6ac91d5d5668da1fa5a2b55303246..2ec1991cef8d023d581b0aeca739fd11bbf276c2 2cf42aa3888e665541faf34a634f054b4580d25f..71e0173d8db17116a1a17517b40512be8c05af50`
